### PR TITLE
Add aliases JSONLines and NDJSON for JSONEachRow

### LINKED ISSUE #36303: Add aliases `JSONLines`, `NDJSON`
To `JSONEachRow`

### PR DESCRIPTION
**Pull request by ucasfl**
### Changelog category (leave one):
- New Feature

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add aliases JSONLines and NDJSON for JSONEachRow. Closes #36303.

### CODE REVIEW

File: src/Processors/Formats/Impl/JSONEachRowRowInputFormat.cpp [hunk 1]
```diff
@@ -360,8 +360,30 @@ void registerInputFormatJSONEachRow(FormatFactory & factory)
...
+        const Block & sample,
+        IRowInputFormat::Params params,
+        const FormatSettings & settings)
+    {
+        return std::make_shared<JSONEachRowRowInputFormat>(buf, sample, std::move(params), settings, false);
+    });
+
     factory.registerFileExtension("ndjson", "JSONEachRow");
     factory.registerFileExtension("jsonl", "JSONEachRow");
+    factory.registerFileExtension("ndjson", "JSONLines");
```
Comment (alexey-milovidov) [inline src/Processors/Formats/Impl/JSONEachRowRowInputFormat.cpp:383]: As far as I remember, file extension is a mapping from extension to data format. It should not be ambiguous.

File: src/Processors/Formats/Impl/JSONEachRowRowInputFormat.cpp [hunk 2]
```diff
@@ -360,8 +360,30 @@ void registerInputFormatJSONEachRow(FormatFactory & factory)
         return std::make_shared<JSONEachRowRowInputFormat>(buf, sample, std::move(params), settings, false);
     });
 
+    factory.registerInputFormat("JSONLines", [](
```
Comment (alexey-milovidov) [inline src/Processors/Formats/Impl/JSONEachRowRowInputFormat.cpp:363]: Does the factory allow registration of aliases?

Comment (ucasfl) [reply src/Processors/Formats/Impl/JSONEachRowRowInputFormat.cpp:363]: I checked it does not support registration of aliases.

**Comment by alexey-milovidov** [comment]
Let's merge with master after #36337

**Comment by ucasfl** [comment]
@Mergifyio update

**Comment by alexey-milovidov** [comment]
Waiting for #36361.

**Comment by alexey-milovidov** [comment]
Unit test `CoordinationTestSuite/CoordinationTest.TestRotateIntervalChanges/0` has failed.

**Comment by alexey-milovidov** [comment]
Unit test `CoordinationTestSuite/CoordinationTest.TestRotateIntervalChanges/0` has failed.

**Comment by ucasfl** [comment]
> Unit test `CoordinationTestSuite/CoordinationTest.TestRotateIntervalChanges/0` has failed.

The failed test seems does not related to my changes.

**Comment by alexey-milovidov** [comment]
Yes, but it has to be analyzed before merging this PR.

**Comment by alexey-milovidov** [comment]
Waiting for @alesapin to fix the unit test.
Or @ucasfl you can help.

**Comment by ucasfl** [comment]
@Mergifyio update

**Pull request merged**